File: TransCoder_tokenized_test_set_functions/java/CHECK_DIVISIBILITY_LARGE_NUMBER_999.java
static boolean isDivisible999 ( String num ) {
  int n = num . length ( );
  if ( n == 0 && num . charAt ( 0 ) == '0' ) return true;
  if ( n % 3 == 1 ) num = "00" + num;
  if ( n % 3 == 2 ) num = "0" + num;
  int gSum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int group = 0;
    group += ( num . charAt ( i ++ ) - '0' ) * 100;
    group += ( num . charAt ( i ++ ) - '0' ) * 10;
    group += num . charAt ( i ) - '0';
    gSum += group;
  }
  if ( gSum > 1000 ) {
    num = Integer . toString ( gSum );
    n = num . length ( );
    gSum = isDivisible999 ( num ) ? 1 : 0;
  }
  return ( gSum == 999 );
}

File: TransCoder_tokenized_test_set_functions/python/CHECK_DIVISIBILITY_LARGE_NUMBER_999.py
def isDivisible999 ( num ) :
    n = len ( num );
    
    if ( n == 0 or num [ 0 ] == '0' ) :
        return true
    if ( ( n % 3 ) == 1 ) :
        num = "00" + num
    if ( ( n % 3 ) == 2 ) :
        num = "0" + num
    gSum = 0
    for i in range ( 0, n, 3 ) :
        group = 0
        group += ( ord ( num [ i ] ) - 48 ) * 100
        group += ( ord ( num [ i + 1 ] ) - 48 ) * 10
        group += ( ord ( num [ i + 2 ] ) - 48 )
        gSum += group
    if ( gSum > 1000 ) :
        num = str ( gSum )
        n = len ( num )
        gSum = isDivisible999 ( num )
    return ( gSum == 999 )
